components/collections/src/codepointtrie/cptrie.rs

```diff
@@ -438,6 +438,44 @@ impl<'trie, T: TrieValue> CodePointTrie<'trie, T> {
         })
     }
 
+    /// Maps the CodePointTrie into one that returns a different type.
+    ///
+    /// This function returns owned data.
+    ///
+    /// If the old and new types are the same size, use the more efficient
+    /// [`CodePointTrie::try_into_converted`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_collections::codepointtrie::CodePointTrie;
+    /// use icu_collections::codepointtrie::planes;
+    /// use core::convert::Infallible;
+    ///
+    /// let planes_trie_u8: CodePointTrie<u8> = planes::get_planes_trie();
+    /// let planes_trie_u16: CodePointTrie<u16> = planes_trie_u8
+    ///     .try_alloc_map_value(TryFrom::try_from)
+    ///     .expect("infallible");
```

Comment:
If https://github.com/rust-lang/rust/issues/61695 ever gets stabilized then we won't need to `.expect("infallible")` any more